app/editor/src/features/admin/sources/DataSourceForm.tsx

```diff
@@ -82,16 +87,34 @@ export const DataSourceForm: React.FC<IDataSourceProps> = (props) => {
           handleSubmit(values);
           setSubmitting(false);
         }}
+        validateOnBlur={true}
```

Comment:
On blur so that validation occurs and is cleared after it has been run.  I don't love this solution either because it runs validation early and the tabs go red if they have required fields.  Once again I don't have a better way to do this presently.